components/SettingsControls/src/Helpers/ControlHelper.cs

```diff
@@ -0,0 +1,9 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace CommunityToolkit.Labs.WinUI;
+internal static partial class ControlHelpers
+{
+    internal static bool IsXamlRootAvailable { get; } = Windows.Foundation.Metadata.ApiInformation.IsPropertyPresent("Windows.UI.Xaml.UIElement", "XamlRoot");
```

Comment:
@michael-hawker This helpers comes back in other controls as well (e.g. TokenView). Is this something we could put into a common library?

Reply:
It's a pain as we target 1809, but the XamlRoot was only added in 1903... I mean it's a simple shortcut, so don't know if worth a common helper and taking a dependency, unless we do like a Source Generator magic sort of thing... 😋 But that's not simple either.

We could also try and up our min SDK, but then we wouldn't be aligned with the WinAppSDK side, though it supports this property... 21H2 is the next LTSC release that's out now.

But less than 2% of folks are on 1809 or older, most are on 1903 and above, latest report I could find from middle of last year: https://reports.adduplex.com/#/r/2022-06

![image](https://user-images.githubusercontent.com/24302614/230164017-459f7cb8-bf28-4127-9a53-668b1eb1ff94.png)

Reply:
For now, let's just keep copying this simpler helper as needed.

@Sergio0694 is PolySharp just for .NET runtime stuff or could it or something similar be used to generate this type of thing as well?

Reply:
Honestly I'd just bump to 1903 (18362) everywhere 😄

Reply:
@Sergio0694 yeah, it's a bit odd as I think WinUI 3 supports down to 1809, so then we'd be mismatched for LTSC for UWP there, that's my only concern really.